library/core/src/slice/iter.rs

```diff
@@ -94,8 +92,7 @@ impl<'a, T> Iter<'a, T> {
         unsafe {
             assume(!ptr.is_null());
```

Comment:
related to your previous PR, is this assume still needed?

Reply:
Probably not?  I wish we had a way to directly get a `NonNull<T>` from a `&[T]`, which would make it rather more obviously unnecessary here.

Do you want me to address this in this PR, or leave it for the future?

Reply:
Leave it.